fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1707,6 +1718,89 @@ private List<Tuple> queryLocal(Map<Tuple, Long> dataset, Comparisons.Type compar
+                                          int[] expectedPartitionCounts) {
+        boolean isSynthetic = false;
+
+        Map<String, String> options = Map.of(
+                INDEX_PARTITION_BY_FIELD_NAME, isSynthetic ? "complex.timestamp" : "timestamp",
+                INDEX_PARTITION_HIGH_WATERMARK, String.valueOf(highWatermark),
+                INDEX_PARTITION_LOW_WATERMARK, String.valueOf(lowWatermark)
+                );
+        Pair<Index, Consumer<FDBRecordContext>> indexConsumerPair = setupIndex(options, true, isSynthetic);
+        Index index = indexConsumerPair.getLeft();
+        Consumer<FDBRecordContext> schemaSetup = indexConsumerPair.getRight();
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, repartitionCount)
+                .build();
+
+        final String luceneSearch = isSynthetic ? "simple_text:about" : "text:vision";
+        Map<Tuple, Map<Tuple, Tuple>> createdKeys;
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            recordStore.getIndexDeferredMaintenanceControl().setAutoMergeDuringCommit(false);
+            createdKeys = createPartitionsAndComplexDocs(index, initialPartitionCounts);
+            context.commit();
+        }
+
+        List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = getPartitionMeta(index,
+                Tuple.from(1), contextProps, schemaSetup);
+        partitionInfos.sort(Comparator.comparing(LucenePartitionInfoProto.LucenePartitionInfo::getId));
+
+        // assert sane initial setup
+        assertEquals(initialPartitionCounts.length, partitionInfos.size());
+        assertArrayEquals(initialPartitionCounts, partitionInfos.stream().mapToInt(LucenePartitionInfoProto.LucenePartitionInfo::getCount).toArray());
+
+        explicitMergeIndex(index, contextProps, schemaSetup);
+
+        partitionInfos = getPartitionMeta(index, Tuple.from(1), contextProps, schemaSetup);
+        partitionInfos.sort(Comparator.comparing(LucenePartitionInfoProto.LucenePartitionInfo::getId));
+
+        // assert correct partition setup after repartitioning
+        assertEquals(expectedPartitionCounts.length, partitionInfos.size());
+        assertArrayEquals(expectedPartitionCounts, partitionInfos.stream().mapToInt(LucenePartitionInfoProto.LucenePartitionInfo::getCount).toArray());
+
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            // finally, assert all docs are accounted for
+            assertIndexEntryPrimaryKeyTuples(createdKeys.get(Tuple.from(1L)).keySet(),
+                    recordStore.scanIndex(COMPLEX_PARTITIONED, groupedTextSearch(COMPLEX_PARTITIONED, luceneSearch, 1), null, ExecuteProperties.newBuilder().build().asScanProperties(false)));
```

Comment:
This doesn't seem to validate that the right documents are in the right partitions, it seems important to run `LuceneIndexTestValidator.validate` on this, to ensure that the `from`/`to` all make sense for the documents contained, and that they are still ordered correctly.

Reply:
see above